### ISSUE TITLE
Non-contiguous log error

### ISSUE THREAD

**Issue by benbjohnson** [influxdb/influxdb#2025]
Test failure after 2.4m inserts:

```
panic: assert failed: non-contiguous log index(1): idx=2896, prev=2896

goroutine 319590 [running]:
github.com/influxdb/influxdb/raft.assert(0xc22533dc00, 0x976270, 0x2d, 0xc22533dc00, 0x3, 0x3)
        /root/go/src/github.com/influxdb/influxdb/raft/log.go:1805 +0xcf
github.com/influxdb/influxdb/raft.(*Log).append(0xc2080384e0, 0xc21a05be30)
        /root/go/src/github.com/influxdb/influxdb/raft/log.go:1156 +0x1f8
github.com/influxdb/influxdb/raft.(*Log).ReadFrom(0xc2080384e0, 0x7fe695685fc0, 0xc217ae33c0, 0x0, 0x0)
        /root/go/src/github.com/influxdb/influxdb/raft/log.go:1702 +0xb29
github.com/influxdb/influxdb/raft.(*Log).readFromLeader(0xc2080384e0, 0xc237a00e20, 0xc238606420)
        /root/go/src/github.com/influxdb/influxdb/raft/log.go:799 +0x60b
created by github.com/influxdb/influxdb/raft.(*Log).followerLoop
        /root/go/src/github.com/influxdb/influxdb/raft/log.go:738 +0x18e
```

Reported by @tcoulter

**Comment by otoolep** [comment]
@otoolep

**Comment by pauldix** [comment]
fixed by #2111
